frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -141,20 +142,34 @@ function buildFieldFilterUiParameter(
   );
   const mappedFields = mappingsForParameter.map(mapping => {
     const { target, card } = mapping;
-    return getTargetFieldFromCard(target, card, metadata);
+    const question = new Question(card, metadata);
```

Comment:
In `getTargetFieldFromCard` there is a check for 
```
if (!card?.dataset_query) {
    return null;
  }
```

I don't think that we should drop it. If CI passed, could you please add relevant e2e tests to make sure this is covered?

Reply:
I'm not sure in what scenario, `card` would be null since `card?.dataset_query` implies that.

Based on where `mapping` is created 
https://github.com/metabase/metabase/blob/e84bd827709a60dd069385144f25f653eaa93e98/frontend/src/metabase/parameters/utils/dashboards.ts#L102-L112

We can clearly see that `card` can never be null (line 104 and 108) so I think the check
```ts
  if (!card?.dataset_query) {
    return null;
  }
```
is already outdated.

Since `getTargetFieldFromCard` is only used in a single place here. I think it's safe to remove that check completely to avoid future confusion. What do you think?

In the mean time I'll try removing `if (!card?.dataset_query)` to see if there's any legit failure or not, but from my investigation that shouldn't be the case.

Reply:
Ahh, I see there is a unit test that is testing that a card doesn't have `dataset_query`. In what scenario this would happen? I don't think it's possible doesn't it?

Reply:
Found the original issue. https://github.com/metabase/metabase/issues/20656 I'll make sure this is covered.

Reply:
This is confirmed not to be a problem because there's already an E2E test to cover that case at `e2e/test/scenarios/dashboard-filters/reproductions/20656-dashboard-breaks-for-user-without-card-permissions.cy.spec.js`

After the removal of that outdated check in https://github.com/metabase/metabase/pull/33632/commits/479871b402407afc14d022e703146653a1dd55f4, the test still passes, so I adjust the unit test to the new behavior instead.